File: game/proj/config_class.lua
require "class"

--local Utf8 = require 'utf8'


config_set = class( function( self )
	self.Configs = {}
    self.ConfigNames = {}
    self.ConfigNameToIndexTab = {} 

	Logf( 'load configs------' )
	local Patch = "res/items/"
	LoadChunkData = love.filesystem.load( Patch .. 'items.lua' ) -- загружает кусок
	LoadChunkData() -- выполняет кусок
		
	for I = 1, #configs_list do
		Logf( 'load ' .. '[' .. I .. ']' .. configs_list[ I ]  )
		LoadChunkData = love.filesystem.load( Patch .. configs_list[ I ] .. '.lua' )
		LoadChunkData()
		self.Configs[ I ] = item_config
		self.ConfigNames[ I ] = configs_list[ I ]
		self.ConfigNameToIndexTab[ configs_list[ I ] ] = I
	end
	Logf( "load configs done, count: " .. #self.Configs )
end)


function config_set:GetByIndex( Index )
	assert( Index > 0 )
	assert( Index <= #self.Configs )
	return self.Configs[ Index ]	
end


function config_set:GetByName( Name )
	return self.Configs[ self.ConfigNameToIndexTab[ Name ] ]
end


function config_set:GetNameByIndex( Index )
	assert( Index > 0 )
	assert( Index <= #self.Configs )
	return self.ConfigNames[ Index ]
end


function config_set:GetIndexByName( Name )
	return self.ConfigNameToIndexTab[ Name ]
end

File: server/game/config_class.py
from helper_class import *
from lib_slpp import slpp as lua



# Start config index = 1
class config_set:
    Configs = None    
    ConfigNames = None
    ConfigNameToIndexTab = None

    def __init__( self ):
        self.Configs = []
        self.ConfigNames = []    
        self.ConfigNameToIndexTab = {}   
        

    def Load( self ):
        ConfigPatch = '../game/proj/res/items/'
  
        print "load configs ----------------"  
               
        with open( ConfigPatch + 'items.lua' ) as data_file: 
            data = lua.decode( '{' + data_file.read() + '}' )            
            Index = 0
            for Patch in data[ 'configs_list' ]:
                 with open( ConfigPatch + Patch + '.lua' ) as config_file: 
                    Config = lua.decode( '{' + config_file.read() + '}' )
                    self.ConfigNames.append( Patch )                    
                    self.Configs.append( Config )
                    print "load config:", Patch 
                    self.ConfigNameToIndexTab[ Patch ] = Index
                    Index += 1
        
        print "load configs done, count:", Index


    def GetByIndex( self, Index ):
        ASSERT( Index > 0 )
        return ( self.Configs[ Index - 1 ] )[ 'item_config' ]


    def GetByName( self, Name ):
        return ( self.Configs[ self.ConfigNameToIndexTab[ Name ] ] )[ 'item_config' ]


    def GetIndexByName( self, Name ):
        return self.ConfigNameToIndexTab[ Name ] + 1


    def GetNameByIndex( self, Index ):
        return self.ConfigNames[ Index - 1 ]


g_Configs = config_set()
